storm-core/src/jvm/backtype/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -0,0 +1,296 @@
+
+  @Override
+  public void prepare(Map conf, String overrideBase, NimbusInfo nimbusInfo) {
+    this.conf = conf;
+    this.nimbusInfo = nimbusInfo;
+     if (overrideBase == null) {
+       overrideBase = (String)conf.get(Config.BLOBSTORE_DIR);
+      if (overrideBase == null) {
+        overrideBase = (String) conf.get(Config.STORM_LOCAL_DIR);
+      }
+    }
+    File baseDir = new File(overrideBase, BASE_BLOBS_DIR_NAME);
+    try {
+      fbs = new FileBlobStoreImpl(baseDir, conf);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    _aclHandler = new BlobStoreAclHandler(conf);
+  }
+
+  @Override
+  public AtomicOutputStream createBlob(String key, SettableBlobMeta meta, Subject who) throws AuthorizationException, KeyAlreadyExistsException {
+    LOG.debug("Creating Blob for key {}", key);
+    validateKey(key);
+    _aclHandler.normalizeSettableBlobMeta(key, meta, who, READ | WRITE | ADMIN);
+    BlobStoreAclHandler.validateSettableACLs(key, meta.get_acl());
+    _aclHandler.hasPermissions(meta.get_acl(), READ | WRITE | ADMIN, who, key);
+    if (fbs.exists(DATA_PREFIX+key)) {
+      throw new KeyAlreadyExistsException(key);
+    }
+    BlobStoreFileOutputStream mOut = null;
+    try {
+      mOut = new BlobStoreFileOutputStream(fbs.write(META_PREFIX+key, true));
+      mOut.write(Utils.thriftSerialize(meta));
+      mOut.close();
+      mOut = null;
+      return new BlobStoreFileOutputStream(fbs.write(DATA_PREFIX+key, true));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } finally {
+      if (mOut != null) {
+        try {
+          mOut.cancel();
+        } catch (IOException e) {
+          //Ignored
+        }
+      }
+    }
+  }
+
+  @Override
+  public AtomicOutputStream updateBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
+    validateKey(key);
+    checkForBlobOrDownload(key);
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+    try {
+      return new BlobStoreFileOutputStream(fbs.write(DATA_PREFIX+key, false));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private SettableBlobMeta getStoredBlobMeta(String key) throws KeyNotFoundException {
+    InputStream in = null;
+    try {
+      LocalFsBlobStoreFile pf = fbs.read(META_PREFIX+key);
+      try {
+        in = pf.getInputStream();
+      } catch (FileNotFoundException fnf) {
+        throw new KeyNotFoundException(key);
+      }
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      byte [] buffer = new byte[2048];
+      int len;
+      while ((len = in.read(buffer)) > 0) {
+        out.write(buffer, 0, len);
+      }
+      in.close();
+      in = null;
+      return Utils.thriftDeserialize(SettableBlobMeta.class, out.toByteArray());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } finally {
+      if (in != null) {
```

Comment:
add a helper method like `closeStreamIfNotNull` to address all similar code blocks? I see a dozen of similar blocks...